### PR TITLE
Post-purchase: Enable user to cancel a migration manually

### PR DESCRIPTION
**Pull request by gabrielcaires**
Related to #96157
Depends on #95963

## Proposed Changes
* Add a button to cancel a pending migration
* Reloads the site after the cancellation. 

> [!IMPORTANT]  
> As demonstrated in the video above, the page header is not listening to the act of reloading the site details, so it shows the pending label until the user reloads the page.  It will be fixed on the issue  #96230 

https://github.com/user-attachments/assets/452564de-59cb-433d-baf5-9fa36092f384

## Testing Instructions
* Enable the feature flag, sessionStorage.setItem('flags', 'automated-migration/pending-status');
* Set a site with one of the pending status (`migration-pending-diy` or `migration-pending-difm`)  using any of the migration flows and stopping on the credential step(DIFM) or the instructions step(DIY)
* Go to `/site` .
* Click on the site with the `pending` label.
* Click on the cancel button.
* Reload the page and check there is no pending label. 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by Imran92 — COMMENTED** [review]
Looks like we have some conflicts.

Also, maybe we should change the text of the Cancel button https://serenityp2.wordpress.com/2024/11/04/post-purchase-experience-marking-pending-migrations/#comment-4295

File: client/hosting/overview/components/migration-overview/components/migration-pending/index.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { Button } from '@wordpress/components';
+import { translate } from 'i18n-calypso';
+import { useCallback, useEffect } from 'react';
+import { HostingHeroButton } from 'calypso/components/hosting-hero';
+import { useMigrationCancellation } from 'calypso/data/site-migration/landing/use-migration-cancellation';
+import { addQueryArgs } from 'calypso/lib/url';
+import { getMigrationType } from 'calypso/sites-dashboard/utils';
+import { useDispatch } from 'calypso/state';
+import { requestSite } from 'calypso/state/sites/actions';
+import Cards from '../cards';
```
Comment (valterlorran) [inline client/hosting/overview/components/migration-overview/components/migration-pending/index.tsx:11]: Nit: I feel this is too generic and would be hard to maintain, I'd suggest relating it with the actual component name, something like MigrationOverviewCards.

File: client/hosting/overview/components/migration-overview/components/migration-started-diy.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { translate } from 'i18n-calypso';
+import Cards from './cards';
+import { Container, Header } from './layout';
+import type { SiteDetails } from '@automattic/data-stores';
+
+export const MigrationStartedDIY = ( { site }: { site: SiteDetails } ) => {
```
Comment (valterlorran) [inline client/hosting/overview/components/migration-overview/components/migration-started-diy.tsx]: We should call out this refactor on the PR description, as I was not expecting to see anything related to the Started status by reading the PR description.

File: client/hosting/overview/components/migration-overview/components/cards.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import { translate } from 'i18n-calypso';
+import { HostingCard, HostingCardGrid } from 'calypso/components/hosting-card';
+
+export const cards = [
```
Comment (valterlorran) [inline client/hosting/overview/components/migration-overview/components/cards.tsx]: I think we need to name this better. I don't feel confident using casing to differentiate constants. If you look below, we are setting Cards. 

Beyond that, I also think we should add a more explicit name, as this is too generic.

**Review by valterlorran — COMMENTED** [review]
The PR is working as described, but there are a few naming issues that I think we should address. We also need to add more context to the PR as, there are things being done that are not described in the description.

**Comment by fditrapani** [comment]
@gabrielcaires did you add the modal to this one yet?

**Comment by gabrielcaires** [comment]
@valterlorran When you reviewed the PR, GitHub was showing a wrong diff. 
The refactoring and changes you mentioned were related to my previous PR, where I introduced the refactor. 

I didn't introduce changes on the card or the card components here.

File: client/hosting/overview/components/migration-overview/components/cards.tsx (hunk 3 above)
Comment (gabrielcaires) [reply client/hosting/overview/components/migration-overview/components/cards.tsx]: Answered here https://github.com/Automattic/wp-calypso/pull/96160#issuecomment-2468373156

File: client/hosting/overview/components/migration-overview/components/migration-started-diy.tsx (hunk 2 above)
Comment (gabrielcaires) [reply client/hosting/overview/components/migration-overview/components/migration-started-diy.tsx]: Answered it here https://github.com/Automattic/wp-calypso/pull/96160#issuecomment-2468373156

File: client/hosting/overview/components/migration-overview/components/migration-pending/index.tsx (hunk 1 above)
Comment (gabrielcaires) [reply client/hosting/overview/components/migration-overview/components/migration-pending/index.tsx:11]: Answered here https://github.com/Automattic/wp-calypso/pull/96160#issuecomment-2468373156

**Comment by gabrielcaires** [comment]
> @gabrielcaires did you add the confirm modal to this one yet?

No, yet, I put it in another card task, #96237

**Comment by fditrapani** [comment]
No problem. Sounds good! Thanks

**Review by valterlorran — APPROVED** [review]
LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17000322

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @gabrielcaires, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Cancel migration`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17000322).

**Pull request merged**